### PR TITLE
Add builder for Curl

### PR DESCRIPTION
**Pull request by giordano**
CC: @DilumAluthge

### CODE REVIEW

**Comment by giordano** [comment]
I guess I could add OpenSSL to the dependencies :thinking:  and zlib, too

File: C/Curl/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Collection of sources required to build Curl
+sources = [
+    "https://curl.haxx.se/download/curl-$(version).tar.bz2" =>
+    "6618234e0235c420a21f4cb4c2dd0badde76e6139668739085a70c4e2fe7a141"
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/curl-*/
+./configure --prefix=$prefix --host=$target --build=${MACHTYPE} --disable-static
```
Comment (giordano) [inline C/Curl/build_tarballs.jl]: I had to specify the build system to convince `configure` that this is a cross compilation :confused:

Comment (DilumAluthge) [reply C/Curl/build_tarballs.jl]: That seems fine to me.

Comment (DilumAluthge) [reply C/Curl/build_tarballs.jl]: Is there something you don't like about that?

**Comment by giordano** [comment]
There is already the builder in libcurl 😩 naming is hard, but I should have grepped it

**Comment by DilumAluthge** [comment]
Happens to the best of us

**Pull request closed without merging**